src/nasher/utils/git.nim

```diff
@@ -38,13 +36,80 @@ proc gitRemote*(dir = getCurrentDir()): string =
         let ssh = parseUri("ssh://" & result)
         result = ("https://$1/$2$3") % [ssh.hostname, ssh.port, ssh.path]
 
-proc gitInit*(dir = getCurrentDir()): bool =
+proc gitInit*(repo = getCurrentDir()): bool =
   ## Initializes dir as a git repository and returns whether the operation was
   ## successful. Will throw an OSError if dir does not exist.
-  withDir(dir):
-     execCmdEx("git init").exitCode == 0
+  withDir(repo):
+    execCmdEx("git init").exitCode == 0
+
+proc empty(repo: string): bool =
+  # Check if repo has any commits
+  withDir(repo):
+    gitExecCmd("git branch --list", "none").len == 0
+
+proc exists(repo: string): bool =
+  # Check for repo existence
+  withDir(repo):
+    gitExecCmd("git rev-parse --is-inside-work-tree") == "true"
+
+proc exists(branch: string, repo: string): bool =
+  # Check for branch existence
+  withDir(repo):
+    gitExecCmd("git show-ref --verify refs/heads/" & branch, "error") != "error"
+
+proc checkout(branch: string, repo: string, create = false): bool = 
+  # Checkout desired branch, if it exists.  If not, prompts for creation or
+  # uses of current branch.  If can't checkout because of an error, do something else?
+  if create:
+    withDir(repo):
+      gitExecCmd("git checkout -b " & branch, "error") != "error"
+  else:
+    withDir(repo):
+      gitExecCmd("git checkout " & branch, "error") != "error"
+
+proc branch(repo: string, default = ""): string =
+  # Gets the current repo branch
+  withDir(repo):
+    gitExecCmd("git rev-parse --abbrev-ref HEAD", default)
+
+proc create(repo: string, branch: string):bool =
+  # Wrapper function for checkout; creates a new git branch in repo
+  branch.checkout(repo, true)
+
+proc gitSetBranch*(repo = getCurrentDir(), branch: string): string =
+  ## Called if the branch option was specified in configuration or command line
+  if repo.exists:
+    if branch.exists(repo):
+      if repo.branch == branch:
+        result = branch
+      else:
+        if branch.checkout(repo):
+          result = repo.branch
+        else:
+          withDir(repo):
+            error(gitExecCmd("git checkout " & branch))
+
+          fatal(fmt"You must resolve the git repo error before you can continue on branch {branch}")
+    else:
+      if repo.empty:
+        let question = "Nasher cannot determine the status of this repo because there have not been any commits. " &
+                       fmt"Continue the operation on branch {branch}?"
+
+        if askIf(question):
+          if repo.create(branch): 
+            if branch != "master":
+              warning("check repo structure, orphan branch may have been created")
+
+            result = branch
+          else: fatal(fmt"branch {branch} could not be created")
+        else:
+          fatal("operation aborted by user")
+      else:
+        if branch.create(repo): result = branch
```

Comment:
This should be `repo.create(branch)`.

Reply:
Yep, used to have `branch.create(repo)`, then figured `repo.create(branch)` read better, but missed one. Good catch!